update_existing_families.py

```diff
@@ -149,8 +160,68 @@ def load_csv_file(filename, expected_headers):
                         'OTHER_DATE': ''
                     }
                     dp_import_studentrecords.append(studentrecord)
+
     if len(donor_ids_encountered) == 0:
-        print("TODO Create record for new donor / new student %s and add to dp_import_studentrecords" % (student_id))
+        # At this point we know that: 
+        #   (a) this student isn't in DP (i.e., this is new student)
+        #   (b) this student doesn't have an sibling currently in BSD
+        # Chances are the donor is also new, unless 
+        #   (i) the donor has a kid that has graduated already OR, 
+        #   (ii) this student is returning after a break from BSD, but somehow was given a new student ID
+        # If either (i) or (ii) is true, then DP will not create a new record, but update the existing donor record, so we should be okay. 
+        # At any rate, we have to prepare a new record for this donor, with several custom fields
+        print("Creating record for new donor w/ new student %s" % (student_id))
+
+        h_f_name = district_record['Parent 1 First Name']
+        w_f_name = district_record['Parent 2 First Name']
+        h_l_name = district_record['Parent1 Last Name']
+        w_l_name = district_record['Parent 2 Last Name']
+        if w_l_name == h_l_name :
+            salutation = h_f_name + " and " + w_f_name + " " + w_l_name
+        else:
+            salutation = h_f_name + " " + h_l_name + " and " + w_f_name + " " + w_l_name
+
+        if len(district_record['Parent1Email']) == 0:
+            email = district_record['Parent2Email']
+        else:
+            email = district_record['Parent1Email']
+            
+            
+        newdonorrecord = {
+                'FIRST_NAME': h_f_name,
+                'LAST_NAME': h_l_name,
+                'SP_FNAME': w_f_name,
+                'SP_LNAME': w_l_name,
+                'SALUTATION': salutation,
+                'INFORMAL_SAL': h_f_name + " " + w_f_name,
```

Comment:
You're missing "and". Also, you're not handling the case where the district data only has 1 parent.